### PR TITLE
Refactor LTI Course Create View

### PR DESCRIPTION
**Pull request by nbuonin**
Also updates templates to render the pages with styles inside Canvas

### CODE REVIEW

File: locustempus/main/views.py [hunk 1]
```diff
@@ -230,91 +230,109 @@ def post(self, request, *args, **kwargs) -> HttpResponse:
...
+        user = self.request.user
         course_context = self.request.POST.get('lms_course')
         title = self.request.POST.get('lms_course_title')
-
-        sis_course_id = self.request.POST.get('sis_course_id', '')
-        d = CanvasTemplate.to_dict(sis_course_id)
-
-        if d:
-            (group, faculty_group) = self.groups_from_sis_course_id(d)
+        sis_course_id = self.request.POST['sis_course_id'] \
```
Comment (sdreher) [inline locustempus/main/views.py]: Why not use `self.request.POST.get('sis_course_id', None)` here?

File: locustempus/main/views.py [hunk 2]
```diff
@@ -230,91 +230,109 @@ def post(self, request, *args, **kwargs) -> HttpResponse:
...
-        self.request.user.groups.add(faculty_group)
+        user.groups.add(faculty_group)
         return (group, faculty_group)
 
-    def get_year_and_term_from_sis_course_id(self, sis_course_id):
-        m = re.match(
+    def add_yt_to_course(self, sis_course_id: str, course: Course) -> None:
+        """
+        Sets the year and term attributes on a course if
+        they can be deteremend from a sis_course_id
```
Comment (sdreher) [inline locustempus/main/views.py]: `determined`

**Comment by sdreher** [comment]
Note: the review had only minor points. I'm happy to merge as is, just let me know.

File: locustempus/main/views.py (hunk 1 above)
Comment (nbuonin) [reply locustempus/main/views.py]: I needed this to coerce the type from a string `'None'` to `None`. TBH I was surprised that Django didn't do this for me. Maybe I'm missing something.
I do think I could make the logic more clear by swapping the clauses, let me try that.

**Pull request merged**